public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,171 @@
+# dvc.api.open()
+
+[Context manager](https://www.python.org/dev/peps/pep-0343/#context-managers-in-the-standard-library)
+to open a tracked file.
+
+## Definition
+
+### Signature
+
+```py
+open(path, repo=None, rev=None, remote=None, mode="r", encoding=None)
+```
+
+### Types
+
+All **parameter** types are
+[string](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str).
+
+The **return** type is a
+[file object](https://docs.python.org/3/glossary.html#term-file-object).
+
+No exceptions are raised by this function directly.
+
+## Description
+
+This function is analogous to the
+[`open()`](https://docs.python.org/3/library/functions.html#open) Python
+builtin, but for files tracked in <abbr>DVC projects</abbr> (by DVC or Git).
+However, it may only be used as a _context manager_ (using the `with` keyword as
+shown in the **Examples** below). There is no `close()`.
+
+> See also `dvc.api.read()` for a shorthand way to read the complete contents of
+> a tracked file.
+
+💡 Note that `dvc.api.open()` is able to
+[stream](https://docs.python.org/3/library/io.html) the file directly from
```

Comment:
I think this link is confusing here ... interface is the same for files we "stream" from memory or disk. I would go with something as simple as. I would not even do a Note out of it - it's very important behavior detail.

For most storages `open` avoid downloading file before making it available - it generates a temporary link and returns an open connection to the storage though which file can be read directly, thus avoiding saving it. Only "..." storages do not support this.

We actually need to create a ticket to add support for those. I can imagine we talk about GDrive?

Reply:
OK, removing stream link, simplified paragraph, and removing "💡 Note" p beginning... ⏳

Reply:
Alright, done. I used your paragraph suggestion (almost). I didn't include "generates a temporary link" because I don't understand what that means. What kind of link? "returns an open connection" seems enough.

> though which file can be read directly

- [ ] Question though: what about writing it? I'm guessing writing is never allowed but I should probably check...

> We actually need to create a ticket to add support for those. I can imagine we talk about GDrive?

Yes, just GDrive. Opened #3408